core/src/main/java/com/novoda/noplayer/AdvertView.java

```diff
@@ -0,0 +1,14 @@
+package com.novoda.noplayer;
+
+import java.util.List;
+
+public interface AdvertView {
+
+    void setup(List<AdvertBreak> advertBreaks, AdvertInteractionListener advertInteractionListener);
```

Comment:
@JozefCeluch, maybe it is better to expose a `getAdvertListener` and hook that up to the other listeners. That way it all becomes push based. We could add another call on the advert listener like `onAdvertsLoaded(List<AdvertBreak>)`. 

Just thinking that `removeMarker` etc is oddly specific, might be better to keep it general.  What do you think?

Reply:
After thinking about it more I was a little worried about the threading if we took this approach but it seems that all listeners are called on the thread that the player is created with. In this case it would be the main thread.